Unable to build cmake config for lldb

Hi I am trying to build lldb to use some of the debugging tools for a dap configuration / adapter but i am stuck / unable to build due to the following issue / error 
`asmodeus@Asmodeus:~/.local/share/llvm-project$ cmake -S llvm -B build -DCMAKE_BUILD_TYPE=Release -DLLVM_ENABLE_PROJECTS="lldb"
-- bolt project is disabled
-- clang project is disabled
-- clang-tools-extra project is disabled
-- compiler-rt project is disabled
-- cross-project-tests project is disabled
-- libc project is disabled
-- libclc project is disabled
-- libcxx project is disabled
-- libcxxabi project is disabled
-- libunwind project is disabled
-- lld project is disabled
-- lldb project is enabled
-- mlir project is disabled
-- openmp project is disabled
-- polly project is disabled
-- pstl project is disabled
-- flang project is disabled
-- Found LibXml2: /usr/lib/x86_64-linux-gnu/libxml2.so (found version "2.9.10")
-- Native target architecture is X86
-- Threads enabled.
-- Doxygen disabled.
-- Go bindings disabled.
-- Could NOT find OCaml (missing: OCAMLFIND OCAML_VERSION OCAML_STDLIB_PATH)
-- Could NOT find OCaml (missing: OCAMLFIND OCAML_VERSION OCAML_STDLIB_PATH)
-- OCaml bindings disabled.
-- LLVM host triple: x86_64-unknown-linux-gnu
-- LLVM default target triple: x86_64-unknown-linux-gnu
-- Building with -fPIC
-- Targeting AArch64
-- Targeting AMDGPU
-- Targeting ARM
-- Targeting AVR
-- Targeting BPF
-- Targeting Hexagon
-- Targeting Lanai
-- Targeting Mips
-- Targeting MSP430
-- Targeting NVPTX
-- Targeting PowerPC
-- Targeting RISCV
-- Targeting Sparc
-- Targeting SystemZ
-- Targeting VE
-- Targeting WebAssembly
-- Targeting X86
-- Targeting XCore
-- Could NOT find LibEdit (missing: LibEdit_INCLUDE_DIRS LibEdit_LIBRARIES)
-- Enable editline support in LLDB: FALSE
-- Enable curses support in LLDB: TRUE
-- Enable LZMA compression support in LLDB: TRUE
-- Could NOT find SWIG (missing: SWIG_EXECUTABLE SWIG_DIR) (Required is at least version "3.0")
-- SWIG 3 or later is required for Lua support in LLDB but could not be found
-- Could NOT find LuaAndSwig (missing: LUA_LIBRARIES LUA_INCLUDE_DIR SWIG_EXECUTABLE)
-- Enable Lua scripting support in LLDB: FALSE
-- Could NOT find SWIG (missing: SWIG_EXECUTABLE SWIG_DIR) (Required is at least version "3.0")
-- SWIG 3 or later is required for Python support in LLDB but could not be found
-- Could NOT find PythonAndSwig (missing: Python3_LIBRARIES Python3_INCLUDE_DIRS SWIG_EXECUTABLE)
-- Enable Python scripting support in LLDB: FALSE
-- Found LibXml2: /usr/lib/x86_64-linux-gnu/libxml2.so (found suitable version "2.9.10", minimum required is "2.8")
-- Enable Libxml 2 support in LLDB: TRUE
-- Enable libfbsdvmcore support in LLDB: 0
-- LLDB version: 15.0.0git
CMake Error at /home/asmodeus/.local/share/nvim/lsp_extensions/ccpp-debug/source/lldb/cmake/modules/LLDBConfig.cmake:281 (message):
  Expected directory for clang-resource headers not found:
Call Stack (most recent call first):
  /home/asmodeus/.local/share/nvim/lsp_extensions/ccpp-debug/source/lldb/CMakeLists.txt:28 (include)


-- Configuring incomplete, errors occurred!
See also "/home/asmodeus/.local/share/nvim/lsp_extensions/ccpp-debug/source/build/CMakeFiles/CMakeOutput.log".
See also "/home/asmodeus/.local/share/nvim/lsp_extensions/ccpp-debug/source/build/CMakeFiles/CMakeError.log".
asmodeus@Asmodeus:~/.local/share/nvim/lsp_extensions/ccpp-debug/source$`

--- a/llvm/CMakeLists.txt
+++ b/llvm/CMakeLists.txt
@@ -181,8 +181,9 @@ endif()
 
 if ("lldb" IN_LIST LLVM_ENABLE_PROJECTS)
   if (NOT "clang" IN_LIST LLVM_ENABLE_PROJECTS)
-    message(FATAL_ERROR "Clang is not enabled, but is required for lldb.")
-  endif ()
+    message(STATUS "Enabling clang as a dependency of lldb")
+    list(APPEND LLVM_ENABLE_PROJECTS "clang")
+  endif()
 endif ()
 
 if ("libc" IN_LIST LLVM_ENABLE_PROJECTS)

--- a/llvm/CMakeLists.txt
+++ b/llvm/CMakeLists.txt
@@ -179,6 +179,12 @@ if ("flang" IN_LIST LLVM_ENABLE_PROJECTS)
   endif ()
 endif()
 
+if ("lldb" IN_LIST LLVM_ENABLE_PROJECTS)
+  if (NOT "clang" IN_LIST LLVM_ENABLE_PROJECTS)
+    message(FATAL_ERROR "Clang is not enabled, but is required for lldb.")
+  endif ()
+endif ()
+
 if ("libc" IN_LIST LLVM_ENABLE_PROJECTS)
   message(WARNING "Using LLVM_ENABLE_PROJECTS=libc is deprecated.  Please use "
     "-DLLVM_ENABLE_RUNTIMES=libc or see the instructions at "

